src/Swarm/TUI/Controller.hs

```diff
@@ -788,18 +788,34 @@ handleREPLEvent = \case
 tabComplete :: AppState -> REPLPrompt -> REPLPrompt
 tabComplete _ p@(SearchPrompt {}) = p
 tabComplete s (CmdPrompt t mms)
-  | (m : ms) <- mms = CmdPrompt (replaceLast m t) (ms ++ [m])
+  | (m : ms) <- mms = CmdPrompt (replacementFunc m t) (ms ++ [m])
+  | hasOpenQuotes t = case strMatches of
+    [] -> CmdPrompt t []
+    [m] -> CmdPrompt (completeWith lastQuotedString m) []
+    (m : ms) -> CmdPrompt (completeWith lastQuotedString m) (ms ++ [m])
```

Comment:
I forgot what the idea was with this completing code.

I would expect it to cycle on the suggestions, which makes me wonder why the "one element" and "more element" branches can not be the same.

Reply:
I've added explanatory comments.